src/main/groovy/org/pih/warehouse/inventory/CycleCountUpdateItemCommand.groovy

```diff
@@ -10,6 +14,24 @@ class CycleCountUpdateItemCommand implements Validateable {
 
     boolean recount
 
+    @BindUsing({ obj, source ->
+        def productId = source['inventoryItem']['product'] instanceof Map ? source['inventoryItem']['product']['id'] : source['inventoryItem']['product']
+        def lotNumber = source['inventoryItem']['lotNumber']
+        def expirationDate = source['inventoryItem']['expirationDate'] ? DateUtil.asDate(source['inventoryItem']['expirationDate'].toString()) : null
+
+        Product product = Product.read(productId)
+        InventoryItem inventoryItem = InventoryItem.findByProductAndLotNumber(product, lotNumber) ?: new InventoryItem(product: product, lotNumber: lotNumber)
```

Comment:
Let's follow the pattern done in the `CycleCountItemCommand`. Given that it's the same logic in this command and in `CycleCountItemCommand`, let's create a util binder to use them in both places.
You can check out for an example here: https://github.com/openboxes/openboxes/blob/3ed4a8492e6c049a1d2b98d45ffb16416e1dc4b1/src/main/groovy/org/pih/warehouse/outbound/ImportPackingListItem.groovy#L26-L27